src/fmgc/src/components/EfisLabels.ts

```diff
@@ -0,0 +1,32 @@
+import { FlightLevel } from '@fmgc/guidance/vnav/verticalFlightPlan/VerticalFlightPlan';
+import { FmgcComponent } from '@fmgc/lib/FmgcComponent';
+import { FlightPlanManager } from '@fmgc/wtsdk';
+
+export class EfisLabels implements FmgcComponent {
+    private lastTransitionAltitude: Feet;
+
+    private lastTransitionLevel: FlightLevel;
+
+    private flightPlanManager: FlightPlanManager;
+
+    init(): void {
+        // FIXME we need a better way to fetch this... maybe just make it a singleton
+        this.flightPlanManager = FlightPlanManager.DEBUG_INSTANCE;
```

Comment:
**suggestion**: If the `FlightPlanManager` is essential to most `FmgcComponent` implementations, you could consider moving the responsibility of providing the instance "up" to `loop.ts`. It would at the very least reduce the amount of FIXME things that will come out of this when more `FmgcComponent` implementations are added.

Reply:
Good suggestion. I think a great many components will indeed require access to the flight plan.